app/javascript/components/domain-form/index.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { API } from '../../http_api';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import schema from './domain-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import handleFailure from '../../helpers/handle-failure';
+
+const getInitialValues = id => API.get(`/api/automate_domains/${id}?attributes=name,description,enabled,source`);
+
+const handleSubmit = (id, values, redirectUrl) => {
+  miqSparkleOn(); // miqRedirectBack turns it off
+  const action = id ? 'edit' : 'create';
+  const flashMessage = sprintf(__('Automate Domain "%s" was %s.'),
+    values.name,
+    id ? 'saved' : 'added');
```

Comment:
Don't try tro save bytes on i18n, it just makes the translations harder and more cumbersome.

```
const flashMessage = sprintf(id ? __('Automate Domain "%s" was saved') : __('Automate Domain "%s" was added'), values.name)
```